test/module/irohad/consensus/yac/supermajority_checker_test.cpp

```diff
@@ -18,96 +22,165 @@ using ::testing::ReturnRefOfCopy;
 
 static logger::Logger log_ = logger::testLog("YacCommon");
 
-class SupermajorityCheckerTest : public ::testing::Test,
-                                 public SupermajorityCheckerImpl {
+static const std::map<ConsistencyModel, unsigned int> kf1_param{
+    {ConsistencyModel::kCft, detail::kSupermajorityCheckerKfPlus1Cft},
+    {ConsistencyModel::kBft, detail::kSupermajorityCheckerKfPlus1Bft}};
+
+class SupermajorityCheckerTest
+    : public ::testing::TestWithParam<ConsistencyModel>,
+      public SupermajorityChecker {
  public:
   void SetUp() override {}
+
+  unsigned int getK() const {
+    return kf1_param.at(GetParam());
+  }
+
+  size_t getAllowedFaultyPeers(size_t total_peers) const {
+    return (total_peers - 1) / getK();
+  }
+
+  size_t getSupermajority(size_t total_peers) const {
+    return total_peers - getAllowedFaultyPeers(total_peers);
+  }
+
+  std::string modelToString() const {
+    return "`" + std::to_string(getK()) + " * f + 1' "
+        + (GetParam() == ConsistencyModel::kBft ? "BFT" : "CFT") + " model";
+  }
+
+  bool hasSupermajority(PeersNumberType current,
+                        PeersNumberType all) const override {
+    return checker->hasSupermajority(current, all);
+  }
+
+  bool canHaveSupermajority(const VoteGroups &votes,
+                            PeersNumberType all) const override {
+    return checker->canHaveSupermajority(votes, all);
+  }
+
+  std::unique_ptr<SupermajorityChecker> checker{
+      getSupermajorityChecker(GetParam())};
 };
 
+using CftAndBftSupermajorityCheckerTest = SupermajorityCheckerTest;
+using BftSupermajorityCheckerTest = SupermajorityCheckerTest;
+
+INSTANTIATE_TEST_CASE_P(Instance,
+                        CftAndBftSupermajorityCheckerTest,
+                        ::testing::Values(ConsistencyModel::kCft,
+                                          ConsistencyModel::kBft),
+                        // empty argument for the macro
+                        );
+
+INSTANTIATE_TEST_CASE_P(Instance,
+                        BftSupermajorityCheckerTest,
+                        ::testing::Values(ConsistencyModel::kBft),
+                        // empty argument for the macro
+                        );
+
 /**
  * @given 2 participants
  * @when check range of voted participants
  * @then correct result
  */
-TEST_F(SupermajorityCheckerTest, SuperMajorityCheckWithSize2) {
-  log_->info("-----------| F(x, 2), x in {0..3} -----------");
-
-  int N = 2;
-  ASSERT_FALSE(checkSize(0, N));
-  ASSERT_FALSE(checkSize(1, N));
-  ASSERT_TRUE(checkSize(2, N));
-  ASSERT_FALSE(checkSize(3, N));
+TEST_P(CftAndBftSupermajorityCheckerTest, SuperMajorityCheckWithSize2) {
+  log_->info("-----------| F(x, 2), x in [0..3] |-----------");
```

Comment:
Probably these logs can be removed, as they were used as test descriptions before given-when-then was introduced.